Explorer/Assets/DCL/SDKComponents/AvatarModifierArea/Systems/AvatarModifierAreaHandlerSystem.cs

```diff
@@ -0,0 +1,153 @@
+        [All(typeof(TransformComponent))]
+        private void SetupAvatarModifierArea(in Entity entity, ref PBAvatarModifierArea pbAvatarModifierArea)
+        {
+            World.Add(entity,
+                new CharacterTriggerAreaComponent(areaSize: pbAvatarModifierArea.Area, targetOnlyMainPlayer: false),
+                new AvatarModifierAreaComponent(pbAvatarModifierArea.ExcludeIds));
+        }
+
+        [Query]
+        [All(typeof(TransformComponent))]
+        private void UpdateAvatarModifierArea(ref PBAvatarModifierArea pbAvatarModifierArea, ref AvatarModifierAreaComponent modifierAreaComponent, ref CharacterTriggerAreaComponent triggerAreaComponent)
+        {
+            if (pbAvatarModifierArea.IsDirty)
+            {
+                pbAvatarModifierArea.IsDirty = false;
+
+                triggerAreaComponent.AreaSize = pbAvatarModifierArea.Area;
+                triggerAreaComponent.IsDirty = true;
+
+                modifierAreaComponent.SetExcludedIds(pbAvatarModifierArea.ExcludeIds);
+
+                // Update effect on now excluded/non-excluded avatars
+                foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { CorrectAvatarHidingState(avatarTransform, modifierAreaComponent.ExcludedIds); }
+            }
+
+            foreach (Transform avatarTransform in triggerAreaComponent.ExitedThisFrame) { ToggleAvatarHiding(avatarTransform, false, modifierAreaComponent.ExcludedIds); }
+
+            foreach (Transform avatarTransform in triggerAreaComponent.EnteredThisFrame) { ToggleAvatarHiding(avatarTransform, true, modifierAreaComponent.ExcludedIds); }
+        }
+
+        [Query]
+        [All(typeof(DeleteEntityIntention), typeof(PBAvatarModifierArea))]
+        private void HandleEntityDestruction(ref CharacterTriggerAreaComponent triggerAreaComponent, ref AvatarModifierAreaComponent modifierComponent)
+        {
+            // Reset state of affected entities
+            foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { ToggleAvatarHiding(avatarTransform, false, modifierComponent.ExcludedIds); }
+        }
+
+        [Query]
+        [None(typeof(DeleteEntityIntention), typeof(PBAvatarModifierArea))]
+        private void HandleComponentRemoval(ref CharacterTriggerAreaComponent triggerAreaComponent, ref AvatarModifierAreaComponent modifierComponent)
+        {
+            // Reset state of affected entities
+            foreach (Transform avatarTransform in triggerAreaComponent.CurrentAvatarsInside) { ToggleAvatarHiding(avatarTransform, false, modifierComponent.ExcludedIds); }
+        }
+
+        internal void ToggleAvatarHiding(Transform avatarTransform, bool shouldHide, HashSet<string> excludedIds)
+        {
+            var found = false;
+
+            // There's no way to do a Query/InlineQuery getting both entity and TransformComponent...
+            globalWorld.Query(in AVATAR_BASE_QUERY,
+                entity =>
```

Comment:
Should be refactored without lambdas on every frame, it allocates much memory

Reply:
As agreed, I will create an issue to refactor that part!